tests/chainer_tests/links_tests/loss_tests/test_crf1d.py

```diff
@@ -0,0 +1,125 @@
+        for b, length in enumerate(self.lengths):
+            for ys in itertools.product(range(self.n_label), repeat=length):
+                z[b] += numpy.exp(self._calc_score(b, ys))
+
+        score = numpy.zeros((self.batches[0],), numpy.float32)
+        for b, length in enumerate(self.lengths):
+            ys = [self.ys[i][b] for i in range(length)]
+            score[b] = self._calc_score(b, ys)
+
+        loss = -(score - numpy.log(z))
+        return numpy.sum(loss) / self.batches[0]
+
+    def setUp(self):
+        self._config_user = chainer.using_config('dtype', self.dtype)
+        self._config_user.__enter__()
+        self.n_label = 3
+
+        self.lengths = [3, 3]
+        self.batches = [2, 2, 2]
+
+        self.xs = [numpy.random.uniform(-1, 1, (b, 3)).astype(self.dtype)
+                   for b in self.batches]
+        self.ys = [numpy.random.randint(
+            0, self.n_label, (b,)).astype(numpy.int32)
+            for b in self.batches]
+
+        self.link = links.CRF1d(n_label=self.n_label)
+        self.cost_shape = (self.n_label, self.n_label)
+
+        if self.dtype == numpy.float16:
+            self.check_forward_options = {'atol': 5e-3}
+        else:
+            self.check_forward_options = {'atol': 1e-4}
+
+    def tearDown(self):
+        self._config_user.__exit__(None, None, None)
+
+    def check_forward(self, x_data, t_data):
+        x = self.link(x_data, t_data)
+        t = self._crf1d(self.link.cost.array, x_data, t_data)
+        testing.assert_allclose(x.data, t,
+                                **self.check_forward_options)
+
+    def test_forward_cpu(self):
+        self.check_forward(self.xs, self.ys)
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.link.to_gpu()
+        self.check_forward(cuda.to_gpu(self.xs), cuda.to_gpu(self.ys))
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'initializer': ['random', None]
+}))
+class TestInitialization(unittest.TestCase):
+
+    def setUp(self):
+        self.n_label = 3
+
+        if self.initializer is None:
+            self.initial_cost = None
+
+        elif self.initializer == 'random':
+            initializer = initializers.GlorotUniform()
+            self.initial_cost = numpy.empty((self.n_label, self.n_label),
+                                            dtype=self.dtype)
+            initializer(self.initial_cost)
+
+        with chainer.using_config('dtype', self.dtype):
+            self.link = links.CRF1d(self.n_label,
+                                    initial_cost=self.initial_cost)
+
+    def check_param(self):
+        link = self.link
+        dtype = self.dtype
+        assert link.cost.dtype == dtype
+
+        if self.initializer is None:
+            cost = numpy.empty(
+                (self.n_label, self.n_label), dtype=self.dtype)
+            initial_cost = initializers.constant.Zero()
```

Comment:
How about setting this in `setUp` as `self.initial_cost` for consistency?

Reply:
Thanks, I follow your suggestion and cleanup a little:bow:.

Reply:
Oh, you should update `check_param` too https://github.com/chainer/chainer/pull/5807#discussion_r243134597